tests/tensor/test_basic.py

```diff
@@ -3838,6 +3839,32 @@ def test_transpose():
     assert ptb.transpose(dmatrix()).name is None
 
 
+def test_matrix_transpose():
+    with pytest.raises(ValueError, match="Input array must be at least 2-dimensional"):
+        ptb.matrix_transpose(np.arange(6))
+
+    x2 = dmatrix("x2")
+    x3 = dtensor3("x3")
+
+    x2v = np.arange(6).reshape((2, 3))
+    x3v = np.arange(12).reshape((2, 3, 2))
+
+    f = pytensor.function(
+        [x2, x3],
+        [
+            x2.mT,
+            ptb.matrix_transpose(x3),
+        ],
+    )
+    t2, t3 = f(x2v, x3v)
+
+    assert equal_computations([t2], [np.transpose(x2v)])
```

Comment:
equal_computations was meant to be used with non evaluated expressions. You shouldn't have any pytensor.function or numpy operations in this test at all.

As an example say you have a helper called foo that return the exponent of the variable, you could write a test like this to make sure the helper is doing what I expected:

```python

x = pt.vector()

res = foo(x)
expected_res = pt.exp(x)
assert equal_computations([res], [expected_res])

Reply:
I understand. How would you suggest to test ```mT```. I does not seem to be callable with just non evaluated expressions.

Reply:
>  I does not seem to be callable with just non evaluated expressions.

What do you mean? Should be exactly the same except instead of `foo(x)` you have `x.foo()`?

Reply:
I was thinking of using ```dmatrix``` and ```dtensor3``` instead of pt.vector(). Does it make sense to do so? (as in the latest commit (ignoring the TODO comment. I'll rectify that))

Reply:
Resolved. Extremely sorry had a problem in tensor/variable.py locally.